components/LikeButton.vue

```diff
@@ -0,0 +1,481 @@
+          </button>
+        </div>
+
+        <div
+          class="like-button-stats"
+          @click="$emit('click-stats')"
+        >
+          <simple-svg
+            :filepath="LikeTextIcon"
+            class="like-button-stats__text-logo"
+            fill="currentColor"
+            stroke="transparent"
+          />
+          <span
+            v-if="totalLike > 0"
+            class="like-button-stats__total-like"
+          >{{ formattedTotalLike }}</span>
+        </div>
+      </div>
+
+    </div>
+  </div>
+</template>
+
+<script>
+import _throttle from 'lodash.throttle';
+
+import ClapEffectIcon from '~/assets/like-button/clap-effect.svg';
+import LikeClapIcon from '~/assets/like-button/like-clap.svg';
+import LikeTextIcon from '~/assets/like-button/like-text.svg';
+
+export default {
+  name: 'like-button',
+  props: {
+    likeCount: {
+      type: Number,
+      default: 0,
+    },
+    totalLike: {
+      type: Number,
+      default: 0,
+    },
+    isSuperLike: {
+      type: [Boolean, String],
+      default: false,
+    },
+  },
+  data() {
+    return {
+      ClapEffectIcon,
+      LikeClapIcon,
+      LikeTextIcon,
+
+      isShowBubble: false,
+      isShowClapEffect: false,
+      isPressingKnob: false,
+      hasMovedKnob: false,
+      lastClientX: 0,
+      clientX: 0,
+      clampedClientX: 0,
+      knobProgress: this.isSuperLike ? 1 : 0,
+    };
+  },
+  computed: {
+    formattedTotalLike() {
+      let { totalLike } = this;
+      let suffix = '';
+      if (totalLike > 1000) {
+        totalLike = Math.floor(totalLike / 1000);
+        suffix = 'k';
+      }
+      return `${totalLike.toLocaleString('en')}${suffix}`;
+    },
+    isLocalSuperLike() {
+      return this.knobProgress === 1;
+    },
+  },
+  watch: {
+    isSuperLike(isSuperLike) {
+      this.knobProgress = isSuperLike ? 1 : 0;
+    },
+  },
+  mounted() {
+    document.addEventListener('mousemove', this.onMovingKnob);
+    document.addEventListener('touchmove', this.onMovingKnob);
+    document.addEventListener('mouseup', this.onReleaseKnob);
+    document.addEventListener('touchend', this.onReleaseKnob);
+    document.addEventListener('click', this.onReleaseKnob);
+  },
+  beforeDestroy() {
+    document.removeEventListener('mousemove', this.onMovingKnob);
+    document.removeEventListener('touchmove', this.onMovingKnob);
+    document.removeEventListener('mouseup', this.onReleaseKnob);
+    document.removeEventListener('touchend', this.onReleaseKnob);
+    document.removeEventListener('click', this.onReleaseKnob);
+  },
+  methods: {
+    setClientX(e) {
+      this.clientX = e.targetTouches ? e.targetTouches[0].clientX : e.clientX;
+    },
+    onClickKnob(e) {
+      if (this.hasMovedKnob) return;
+
+      if (this.isLocalSuperLike) {
+        this.$emit('super-like', e, true);
+      } else {
+        this.isShowBubble = true;
+        setTimeout(() => {
+          this.isShowBubble = false;
+        }, 500);
```

Comment:
should store timers in a data/data array, and `clearTimeout` on `beforeDestroy ` if possible